src/Build/Instance/TaskRegistry.cs

```diff
@@ -635,11 +642,26 @@ private static bool IsTaskFactoryClass(Type type, object unused)
             return relevantTaskRegistrations;
         }
 
+        // Create another set containing architecture-specific task entries.
+        // Then when we look for them, check if the name exists in that.
+        Dictionary<string, RegisteredTaskRecord> overriddenTasks = new Dictionary<string, RegisteredTaskRecord>();
```

Comment:
You're keying off of a `string` name--how does that work with the partial-class-name resolution that normal task lookup does? Does this require an exact match to the invocation?

Reply:
That's a wrinkle in this solution, at the moment it requires an exact match at the calling site. Another aspect to consider, how should this play with the overridetasks file?

Reply:
How hard would it be to do "if the task comes from .overridetasks _and_ there's something elsewhere with an override, error"?

Reply:
Found an area that documentation will need to be updated (if we change it): https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-tasks?view=vs-2022#overridden-tasks

>Tasks in these files override any other tasks with the same names, including tasks in the project file.

Looks like precedence will be:
1. Project UsingTask with Override attribute
2. Anything in .overridetasks (or .tasks?)
3. Project files

> How hard would it be to do "if the task comes from .overridetasks and there's something elsewhere with an override, error"?

If we tried to go that route, one issue would be the lack of state in this class. When parsing a .tasks or .overridetasks file (see `RegisterOverrideTasks`), it calls `LoadAndRegisterFromTasksFile` which calls the static `RegisterTasksFromUsingTaskElement`. I guess we could try and determine whether we're in a .tasks or .overridetasks based on the `IFileSystem` that gets passed? Doesn't look like we can get the current file through it though.